compiler/src/dmd/globals.d

```diff
@@ -82,6 +82,16 @@ enum CLIIdentifierTable : ubyte
     All      = 4, /// The least restrictive set of all other tables
 }
 
+/// When a non-normalized identifier is found, what should happen?
+enum CLIIdentifierNormalization : ubyte
+{
+    default_       = 0, /// The default normalization strategy
+    SilentlyAccept = 1, /// Silently accept the bad string
+    Deprecate      = 2, /// Emit a deprecation but allow
+    Warning        = 3, /// Emit a warning but allow
+    Error          = 4  /// Emit an error and fail
+}
```

Comment:
Why are these needed at all? Just pick one.

Reply:
The one that will be picked is normalization. Which doesn't exist currently (next PR).

People may want to know when an identifier isn't normalized (even in its incomplete quick check algorithm form). We may also not want to switch straight into a warning from silently accept (current default).

If you want me to forego the CLI argument I can. But for now it would be silently accept and that means the quick check algorithm won't be testable for this PR.

Also ImportC is C11, so it would need a different default than to D. Unless you want to break from that (which we can do).

Reply:
Deprecate and error have been removed.

Warning handles the non-silent accept.

Reply:
I'm baffled why C23 would break C11.

Reply:
https://symbl.cc/en/1F522/ is in C11's table.

It is not in XID_Continue which is the source of the continue characters for C23.
To put it simply, C23 is no longer defining the ranges, the Unicode Annex 31 and Unicode standard is.

https://unicode.org/reports/tr31/#D1

Basically, characters are no longer being picked by a person individually, which means things that were once in there that should never have been, are gone. Bye-bye emojis.

They are being picked based on their General Category. Which is their purpose, how it is displayed ext.